bluemira/equilibria/plotting.py

```diff
@@ -664,6 +724,643 @@ def plot_plasma_coil(self):
+            )
+        else:
+            self.coilset_psi = diff_coilset_psi
+
+        diff_plasma_psi = self.ref_plasma_psi - self.eq.plasma.psi()
+        # if all zeros
+        if not np.all(diff_plasma_psi):
+            self.plasma_psi = None
+        elif self.diag_ops.psi_diff in PsiPlotType.PSI_ABS_DIFF:
+            self.plasma_psi = np.abs(diff_plasma_psi)
+        elif self.diag_ops.psi_diff in PsiPlotType.PSI_REL_DIFF:
+            self.plasma_psi = np.abs(diff_plasma_psi) / np.max(np.abs(diff_plasma_psi))
+        else:
+            self.plasma_psi = diff_plasma_psi
+
+        diff_total_psi = self.ref_total_psi - self.eq.psi()
+        # if all zeros
+        if not np.all(diff_total_psi):
+            self.total_psi = None
+        elif self.diag_ops.psi_diff in PsiPlotType.PSI_ABS_DIFF:
+            self.total_psi = np.abs(diff_total_psi)
+        elif self.diag_ops.psi_diff in PsiPlotType.PSI_REL_DIFF:
+            self.total_psi = np.abs(diff_total_psi) / np.max(np.abs(diff_total_psi))
+        else:
+            self.total_psi = diff_total_psi
+
+    def _clean_plots(self):
+        if self.i == 0 and (self.diag_ops.psi_diff in PsiPlotType.DIFF):
+            if self.diag_ops.split_psi_plotsis is EqSubplots.XZ_COMPONENT_PSI:
+                self.cax1 = make_axes_locatable(self.ax[0]).append_axes(
+                    "right", size="5%", pad="2%"
+                )
+                self.cax2 = make_axes_locatable(self.ax[1]).append_axes(
+                    "right", size="5%", pad="2%"
+                )
+            else:
+                self.cax = make_axes_locatable(self.ax).append_axes(
+                    "right", size="5%", pad="2%"
+                )
+        else:
+            if self.diag_ops.psi_diff in PsiPlotType.DIFF:
+                if self.diag_ops.split_psi_plots is EqSubplots.XZ_COMPONENT_PSI:
+                    self.cax1.clear()
+                    self.cax2.clear()
+
+                else:
+                    self.cax.clear()
+
+            if self.diag_ops.split_psi_plots is EqSubplots.XZ_COMPONENT_PSI:
+                for _ax in self.ax:
+                    _ax.clear()
+                    if legend := _ax.get_legend():
+                        legend.remove()
+            else:
+                self.ax.clear()
+                if legend := self.ax.get_legend():
+                    legend.remove()
+
+    def update_plot(self):
+        self._calculate_psi()
+        self._clean_plots()
+
+        # update the plot
+        if self.diag_ops.split_psi_plots is EqSubplots.XZ_COMPONENT_PSI:
+            self.plot_psi_coilset()
+            self.plot_psi_plasma()
+            legend = self.ax[0].legend()
+            legend.set_zorder(10)
+            legend = self.ax[1].legend()
+            legend.set_zorder(10)
+            self.ax[0].set_xlabel("$x$ [m]")
+            self.ax[0].set_ylabel("$z$ [m]")
+            self.ax[0].set_title("Coilset")
+            self.ax[0].set_aspect("equal")
```

Comment:
This seems to be repeated quite a few times across these changes can we abstract it into some helper function? (with title as input?)